File: Samples/Verbum/how-to-embed-a-barcode-into-a-document-e20025-11.1.6-/VB/Extensions/BarCode/BarCodeGeneratorFactory.vb
Imports Microsoft.VisualBasic
	Imports System
	Imports System.Collections.Generic
	Imports DevExpress.XtraPrinting.BarCode
	Imports DevExpress.XtraPrinting.BarCode.Native
	Imports DevExpress.XtraRichEdit.Fields
Namespace BizPad

	Friend NotInheritable Class BarCodeGeneratorFactory

		Private Shared defaultCodeType As Type = GetType(CodabarGenerator)
		Private Shared hashTable As New Dictionary(Of String, Type)()
		Private Sub New()
		End Sub
		Shared Sub New()
			hashTable("codabar") = GetType(CodabarGeneratorHelper)
			hashTable("code128") = GetType(Code128GeneratorHelper)
			hashTable("128") = GetType(Code128GeneratorHelper)
			hashTable("industrial2of5") = GetType(Industrial2of5GeneratorHelper)
			hashTable("2of5") = GetType(Industrial2of5GeneratorHelper)
			hashTable("interleaved2of5") = GetType(Interleaved2of5GeneratorHelper)
			hashTable("2of5i") = GetType(Interleaved2of5GeneratorHelper)
			hashTable("code39") = GetType(Code39GeneratorHelper)
			hashTable("39") = GetType(Code39GeneratorHelper)
			hashTable("3of9") = GetType(Code39GeneratorHelper)
			hashTable("code39extended") = GetType(Code39ExtendedGeneratorHelper)
			hashTable("code39e") = GetType(Code39ExtendedGeneratorHelper)
			hashTable("39e") = GetType(Code39ExtendedGeneratorHelper)
			hashTable("3of9e") = GetType(Code39ExtendedGeneratorHelper)
			hashTable("code93") = GetType(Code93GeneratorHelper)
			hashTable("93") = GetType(Code93GeneratorHelper)
			hashTable("code93e") = GetType(Code93ExtendedGeneratorHelper)
			hashTable("code93e") = GetType(Code93ExtendedGeneratorHelper)
			hashTable("93e") = GetType(Code93ExtendedGeneratorHelper)
			hashTable("code11") = GetType(Code11GeneratorHelper)
			hashTable("11") = GetType(Code11GeneratorHelper)
			hashTable("msi") = GetType(CodeMSIGeneratorHelper)
			hashTable("plessey") = GetType(CodeMSIGeneratorHelper)
			hashTable("postnet") = GetType(PostNetGeneratorHelper)
			hashTable("post") = GetType(PostNetGeneratorHelper)
			hashTable("ean13") = GetType(EAN13GeneratorHelper)
			hashTable("ean8") = GetType(EAN8GeneratorHelper)
			hashTable("ean128") = GetType(EAN128GeneratorHelper)
			hashTable("upca") = GetType(UPCAGeneratorHelper)
			hashTable("upcsupplemental2") = GetType(UPCSupplemental2GeneratorHelper)
			hashTable("upcs2") = GetType(UPCSupplemental2GeneratorHelper)
			hashTable("upcsupplemental5") = GetType(UPCSupplemental5GeneratorHelper)
			hashTable("upcs5") = GetType(UPCSupplemental5GeneratorHelper)
			hashTable("upce0") = GetType(UPCE0GeneratorHelper)
			hashTable("upce1") = GetType(UPCE1GeneratorHelper)
			hashTable("matrix2of5") = GetType(Matrix2of5GeneratorHelper)
			hashTable("2of5m") = GetType(Matrix2of5GeneratorHelper)
			hashTable("pdf417") = GetType(PDF417GeneratorHelper)
			hashTable("datamatrix") = GetType(DataMatrixGeneratorHelper)
			hashTable("ecc200") = GetType(DataMatrixGeneratorHelper)
		End Sub

		Friend Shared Function Create(ByVal symbologyCode As String, ByVal switches As InstructionCollection) As IBarCodeGenerator
			Dim type As Type = Nothing
			If (Not String.IsNullOrWhiteSpace(symbologyCode)) Then
				If (Not hashTable.TryGetValue(symbologyCode.Trim().ToLower(), type)) Then
					type = Nothing
				End If
			End If
			If type Is Nothing Then
				type = defaultCodeType
			End If
			Dim generator As IBarCodeGenerator = CType(Activator.CreateInstance(type), IBarCodeGenerator)
			generator.Initialize(switches)
			Return generator
		End Function
	End Class
End Namespace

File: Samples/Verbum/how-to-embed-a-barcode-into-a-document-e20025-11.1.6-/CS/Extensions/BarCode/BarCodeGeneratorFactory.cs
namespace BizPad {
    using System;
    using System.Collections.Generic;
    using DevExpress.XtraPrinting.BarCode;
    using DevExpress.XtraPrinting.BarCode.Native;
    using DevExpress.XtraRichEdit.Fields;

    internal static class BarCodeGeneratorFactory {

        static Type defaultCodeType = typeof(CodabarGenerator);
        static Dictionary<string, Type> hashTable = new Dictionary<string, Type>();
        static BarCodeGeneratorFactory() {
            hashTable["codabar"] = typeof(CodabarGeneratorHelper);            
            hashTable["code128"] = typeof(Code128GeneratorHelper);
            hashTable["128"] = typeof(Code128GeneratorHelper);            
            hashTable["industrial2of5"] = typeof(Industrial2of5GeneratorHelper);
            hashTable["2of5"] = typeof(Industrial2of5GeneratorHelper);
            hashTable["interleaved2of5"] = typeof(Interleaved2of5GeneratorHelper);
            hashTable["2of5i"] = typeof(Interleaved2of5GeneratorHelper);            
            hashTable["code39"] = typeof(Code39GeneratorHelper);
            hashTable["39"] = typeof(Code39GeneratorHelper);
            hashTable["3of9"] = typeof(Code39GeneratorHelper);
            hashTable["code39extended"] = typeof(Code39ExtendedGeneratorHelper);
            hashTable["code39e"] = typeof(Code39ExtendedGeneratorHelper);
            hashTable["39e"] = typeof(Code39ExtendedGeneratorHelper);
            hashTable["3of9e"] = typeof(Code39ExtendedGeneratorHelper);            
            hashTable["code93"] = typeof(Code93GeneratorHelper);
            hashTable["93"] = typeof(Code93GeneratorHelper);
            hashTable["code93e"] = typeof(Code93ExtendedGeneratorHelper);
            hashTable["code93e"] = typeof(Code93ExtendedGeneratorHelper);
            hashTable["93e"] = typeof(Code93ExtendedGeneratorHelper);            
            hashTable["code11"] = typeof(Code11GeneratorHelper);
            hashTable["11"] = typeof(Code11GeneratorHelper);            
            hashTable["msi"] = typeof(CodeMSIGeneratorHelper);
            hashTable["plessey"] = typeof(CodeMSIGeneratorHelper);
            hashTable["postnet"] = typeof(PostNetGeneratorHelper);
            hashTable["post"] = typeof(PostNetGeneratorHelper);            
            hashTable["ean13"] = typeof(EAN13GeneratorHelper);
            hashTable["ean8"] = typeof(EAN8GeneratorHelper);
            hashTable["ean128"] = typeof(EAN128GeneratorHelper);
            hashTable["upca"] = typeof(UPCAGeneratorHelper);
            hashTable["upcsupplemental2"] = typeof(UPCSupplemental2GeneratorHelper);
            hashTable["upcs2"] = typeof(UPCSupplemental2GeneratorHelper);
            hashTable["upcsupplemental5"] = typeof(UPCSupplemental5GeneratorHelper);
            hashTable["upcs5"] = typeof(UPCSupplemental5GeneratorHelper);
            hashTable["upce0"] = typeof(UPCE0GeneratorHelper);
            hashTable["upce1"] = typeof(UPCE1GeneratorHelper);            
            hashTable["matrix2of5"] = typeof(Matrix2of5GeneratorHelper);
            hashTable["2of5m"] = typeof(Matrix2of5GeneratorHelper);            
            hashTable["pdf417"] = typeof(PDF417GeneratorHelper);
            hashTable["datamatrix"] = typeof(DataMatrixGeneratorHelper);
            hashTable["ecc200"] = typeof(DataMatrixGeneratorHelper);
        }

        internal static IBarCodeGenerator Create(string symbologyCode, InstructionCollection switches) {
            Type type = null;
            if (!String.IsNullOrWhiteSpace(symbologyCode)) {
                if (!hashTable.TryGetValue(symbologyCode.Trim().ToLower(), out type)) {
                    type = null;
                }
            }
            if (type == null) {
                type = defaultCodeType;
            }
            IBarCodeGenerator generator = (IBarCodeGenerator)Activator.CreateInstance(type);
            generator.Initialize(switches);
            return generator;
        }
    }
}